src/tokens.ts

```diff
@@ -40,27 +41,58 @@ function getTokenType(value: string) {
   }
 }
 
+function range(token: AntlrToken): [number, number] {
+  return [token.start, token.stop + 1]
+}
+
+function loc(token: AntlrToken): Location {
+  const textInLines = token.text?.split(/\r?\n/) ?? ['']
```

Comment:
Personal preference, feel free to ignore.

```suggestion
  const tokenText = token.text ?? ''
  const textInLines = tokenText.split(/\r?\n/)
```